code/modules/organs/internal/heart.dm

```diff
@@ -102,7 +100,7 @@
 			pulse++
 
 	// So does SCP-3349
-	if(pulse != PULSE_NORM && scp3349_induced)
+	if(pulse != PULSE_NORM && ((SCP ? SCP.designation : "") == "3349-1"))
```

Comment:
why ternary here?

Reply:
While all hearts have the SCP component, it is only initialized for hearts that are affected by 3349 (i.e., the hearts themselves becoming SCP instances), meaning that without this check, the code will throw runtimes.